File: 01-Creational/1.1 Abstract Factory/csharp/DoorsLimited.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AbstractFactory
{
    class DoorsLimited : GenericCompany
    {
        public Door manfuatureDoor(string brand, int quantity)
        {
            if (brand.Equals("VW"))
            {
                return new VWDoor(quantity);
            }
            else if (brand.Equals("Ford"))
            {
                return new FordDoor(quantity);
            }
            else if (brand.Equals("Audi"))
            {
                return new AudiDoor(quantity);
            }
            else if (brand.Equals("BMW"))
            {
                return new BMWDoor(quantity);
            }
            else return null;
        }

        public Bumper manufactureBumper(string brand, int quantity)
        {
            throw new NotImplementedException();
        }

        public void orderParts(string brand, int quantity)
        {
            Door parts;

            parts = manfuatureDoor(brand, quantity);

            parts.bringPiecesFromInventory();
            parts.startAssemblyLine();
            parts.fillBatch();
            parts.cleanAssemblyLine();
        }
    }
}

File: 01-Creational/1.1 Abstract Factory/c++/DoorsLimited.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   DoorsLimited.cpp
 * Author: casertillo
 * 
 * Created on 30 January 2017, 20:09
 */

#include "DoorsLimited.h"
#include "Door.h"
#include "Bumper.h"
#include "AudiDoor.h"
#include "VWDoor.h"
#include "BMWDoor.h"
#include "FordDoor.h"
#include "cstring"
#include "iostream"

void DoorsLimited::orderParts(char* brand, int quantity){
    Door* parts;
    
    this->brand = brand;
    this->quantity = quantity;
    
    parts = manufactureDoor();
    parts->bringPiecesFromInventory();
    parts->startAssemblyLine();
    parts->fillBatch();
    parts->cleanAssemblyLine();

}
Door* DoorsLimited::manufactureDoor(){
    if(strcmp(this->brand,"VW") == 0)
    {
       return new VWDoor(this->quantity); 
    }else if(strcmp(this->brand,"Ford") == 0)
    {
        return new FordDoor(this->quantity);
    }else if(strcmp(this->brand,"Audi") == 0)
    {
        return new AudiDoor(this->quantity);
    }else if(strcmp(this->brand,"BMW") == 0)
    {
        return new BMWDoor(this->quantity);
    }
}
Bumper* DoorsLimited::manufactureBumper(){

}
